src/solver/conv_asm_1x1u.cpp

```diff
@@ -307,11 +354,150 @@ bool PerformanceConfigConvAsm1x1U::IsValid(const ProblemDescription& problem) co
+        if(!(n_mult <= total_n_blocks))
+            return false;
+    }
+    const auto in_gprs =
+        (chunks_per_wave * n_mult * c_mult + elements_in_dword - 1) / elements_in_dword;
+    const auto acc_gprs = chunks_per_wave * n_mult * k_mult;
+    // TODO last vgpr only for old card.
+    // ADD if(option.machine_version_major == 9)
+    // vgprs  = 4 + 2 * in_gprs + acc_gprs + (img_hw % elements_in_dword != 0 ? 1: 0);
+    // else
+    const auto vgprs = 4 + 2 * in_gprs + acc_gprs + (img_hw % elements_in_dword != 0 ? 1 : 0) + 1;
+    if(c_mult != -1)
+    {
+        if((c_mult % elements_in_dword) != 0)
+            return false;
+        if(!(vgprs < 256))
+            return false;
+        const auto sgprs = 25 + 2 * k_mult * c_mult;
+        if(!(sgprs < 102)) /// \todo This is valid for Gfx8 and Gfx9. Check for newer parts.
+            return false;
+    }
+    if(waves_c_in_group != -1)
+    {
+        if(!(waves_c_in_group <= problem.n_inputs))
+            return false;
+        const int c_per_wave      = (problem.n_inputs + waves_c_in_group - 1) / waves_c_in_group;
+        const int c_per_last_wave = problem.n_inputs - (c_per_wave * (waves_c_in_group - 1));
+        if(c_per_wave % c_mult != 0 || c_per_last_wave % c_mult != 0)
+            return false;
+        // return (c_per_wave % c_mult == 0) && (c_per_last_wave % c_mult == 0);
+    }
+    if(waves_k_in_group != -1)
+    {
+        if(!(k_mult * waves_k_in_group <= problem.n_outputs))
+            return false;
+        if(!(waves_c_in_group * waves_k_in_group <= 16))
+            return false;
+        const auto max_waves_per_CU = (256 / vgprs) * 4;
+        if(!(max_waves_per_CU >= waves_c_in_group * waves_k_in_group))
+            return false;
+    }
+    return true;
+}
+
+bool PerformanceConfigConvAsm1x1U::TryToken(int index, int value, const ProblemDescription& problem)
+{
+    switch(index)
+    {
+    case 0: read_size = value; break;
+    case 1: k_mult = value; break;
+    case 2: chunks_per_wave = value; break;
+    case 3: chunk_size = value; break;
+    case 4: n_mult = value; break;
+    case 5: c_mult = value; break;
+    case 6: waves_c_in_group = value; break;
+    case 7: waves_k_in_group = value; break;
+    default: return false;
+    }
+    return this->IsValidDynamic(problem);
+}
+
+bool IsModelApplicable(const ConvolutionContext& ctx, const ProblemDescription& problem)
+{
+#if MIOPEN_ENABLE_AI_HEUR
+    if(miopen::IsEnabled(MIOPEN_DEBUG_CONV_ASM_1X1U_HEUR{}))
+        return false;
+    if(ctx.GetStream().GetDeviceName() != "gfx908")
+        return false;
+    if(problem.kernel_stride_h != 1)
+        return false;
+#else
+    std::ignore = ctx;
+    std::ignore = problem;
+#endif
+    return true;
+}
+
+std::vector<float> TransformFeatures(const ProblemDescription& problem, const int& n)
+{
+    std::vector<float> features(static_cast<unsigned long>(n * n), 0.0);
```

Comment:
```suggestion
    std::vector<float> features(static_cast<std::size_t>(n * n), 0.0);
```
std::size_t may not be an unsigned long.